web/src/app/api/[...path]/route.ts

```diff
@@ -0,0 +1,124 @@
+import { NextRequest, NextResponse } from "next/server";
+
+/* NextJS is annoying and makes use use a separate function for 
+each request type >:( */
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function POST(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PUT(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function DELETE(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function HEAD(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function OPTIONS(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+async function handleRequest(request: NextRequest, path: string[]) {
+  if (process.env.NODE_ENV !== "development") {
+    return NextResponse.json(
+      {
+        message:
+          "This API is only available in development mode. In production, something else (e.g. nginx) should handle this.",
+      },
+      { status: 404 }
+    );
+  }
+
+  try {
+    const backendUrl = new URL(`http://localhost:8080/${path.join("/")}`);
+    const requestBody = request.body
+      ? JSON.stringify(await request.json())
+      : null;
+    // console.log(requestBody)
+
+    // Get the URL parameters from the request
+    const urlParams = new URLSearchParams(request.url.split("?")[1]);
+
+    // Append the URL parameters to the backend URL
+    urlParams.forEach((value, key) => {
+      backendUrl.searchParams.append(key, value);
+    });
+
+    const response = await fetch(backendUrl, {
+      method: request.method,
+      headers: request.headers,
+      body: requestBody,
+    });
+
+    // Check if the response is a stream
+    if (
+      response.headers.get("Transfer-Encoding") === "chunked" ||
+      response.headers.get("Content-Type")?.includes("stream")
+    ) {
+      // If it's a stream, create a TransformStream to pass the data through
+      const { readable, writable } = new TransformStream();
+      response.body?.pipeTo(writable);
+
+      return new NextResponse(readable, {
+        status: response.status,
+        headers: response.headers,
+      });
+    } else {
+      // If it's not a stream, handle it as before
+      const data = await response.text();
```

Comment:
Don't think it's valid to assume we always want .text() - (e.g. for `204` or any other content types)

Reply:
Added handling for images. I believe 204 should be handled properly (.text() will just be empty string)